packages/components/pie-button/src/decorators.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * A decorator for specifying a list of valid values for a property.
+ * If this property's setter is called with an invalid value, an error is logged and the default value will be assigned instead.
+ * @param validValues - The array of valid values
+ * @param defaultValue - The value to fall back on
+ * @returns
+ */
+export const validValues = (validValues: any[], defaultValue: any) => {
```

Comment:
Not sure if it's a bit confusing having the decorator function and param use the same name here? I'd probably call the decorator `validateValues`